### PR TITLE
.NET 7 (Preview 6) changes and additions

### PR DESCRIPTION
**Pull request by jaffinito**
## Description

Adds .NET 7 apps and build targets for our various integration tests.

**This is currently targetting Preview 6**

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)
- [ ] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

### CODE REVIEW

File: tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/MultiFunctionApplicationHelpers.csproj [hunk 1]
```diff
@@ -28,8 +28,10 @@
     <PackageReference Include="log4net.Ext.Json" Version="2.0.9.1" Condition="'$(TargetFramework)' == 'netcoreapp3.1'" />
     <PackageReference Include="log4net" Version="2.0.12" Condition="'$(TargetFramework)' == 'net5.0'" />
     <PackageReference Include="log4net.Ext.Json" Version="2.0.10.1" Condition="'$(TargetFramework)' == 'net5.0'" />
-    <PackageReference Include="log4net" Version="2.0.14" Condition="'$(TargetFramework)' == 'net6.0'" />
+    <PackageReference Include="log4net" Version="2.0.13" Condition="'$(TargetFramework)' == 'net6.0'" />
```
Comment (JcolemanNR) [inline tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/MultiFunctionApplicationHelpers.csproj:31]: cool, adding back some test coverage? :)

File: tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/NetStandardLibraries/RabbitMQ.cs [hunk 2]
```diff
@@ -2,7 +2,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 // See this project's .csproj file for target framework => RabbitMQ.Client version mappings
-#if NETCOREAPP3_1 || NET48 || NET5_0 || NET6_0
+#if NETCOREAPP3_1 || NET48 || NET5_0 || NET6_0 || NET7_0
```
Comment (JcolemanNR) [inline tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/NetStandardLibraries/RabbitMQ.cs:5]: I don't think we need to itemize all the versions of .NET/.NET Core here anymore after dropping support for .NET Core 2.0/2.1/3.0
```suggestion
#if NETCOREAPP || NET48
```

**Review by JcolemanNR — CHANGES_REQUESTED** [review]
Looks pretty good! I think to merge this we need to add some steps to the all solutions build to install the .NET 7 SDK in the build/test steps. Another concern is that the majority of our test coverage just moved from .NET 6 to .NET 7, but only the logging tests were updated to restore .NET 6 test coverage. Before merging, I think we should restore more .NET 6 test coverage (especially when consolemf is involved)

**Comment by nr-ahemsath** [comment]
We are keeping this draft PR open as we continue to work on making sure that we will support .NET 7 when it goes GA.

**Comment by nr-ahemsath** [comment]
> Looks pretty good! I think to merge this we need to add some steps to the all solutions build to install the .NET 7 SDK in the build/test steps.

I made the change to install the .NET 7 preview SDK, but it looks like the version of Visual Studio installed in the `windows-2022` runner (17.3.something) won't pick it up.  This jives with how I had to install Visual Studio 17.4.1 Preview in order to use the .NET 7 Preview 7 SDK on my local system.  I did search for how to install a preview version of Visual Studio in a GHA runner but didn't find anything.

To jog my memory, I looked at what was done last year for .NET 6, and found this PR: https://github.com/newrelic/newrelic-dotnet-agent/pull/790/files.  What's notable is that that PR was merged on 11/4/21, whereas the official GA date for .NET 6 was 11/8/21.  So, it may be that once we get to release candidates for .NET 7, a newer version of Visual Studio and MSBuild will be available and things will start working.

**Comment by nr-ahemsath** [comment]
Since we merged the other .NET 7 branch, which was initially branched off of this one, this can be closed.

**Pull request closed without merging**